tracer/tracer.go

```diff
@@ -465,6 +464,41 @@ func removeTemporaryMaps(ebpfMaps map[string]*cebpf.Map) error {
 	return nil
 }
 
+// rewriteMaps replaces all references to named eBPF maps.
+// This means that pre-existing maps are used, instead of new ones created
+// when calling NewCollection. Any named eBPF maps are removed from CollectionSpec.Maps.
+//
+// Returns an error if a named eBPF map isn't used in at least one program.
```

Comment:
Cilium apparently deprecated this because the FDs from `*cebpf.Map` are taken, but no references to the Map entries are kept. The API issue is that, if no one else holds a reference to these `Map` entries, they can get GCd, the FD closed and the program load fails.

I believe we hold the whole `map` somewhere else, so this should not be an issue. But perhaps this should be documented here that the caller must retain refence to the used `cebpf.Map` objects or the collection load will fai.

Reply:
Updated the comment with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/966/commits/17f6a1583ae972eb574e6b77cdcc83a21abce4f8